src/sentry/api/endpoints/organization_release_details.py

```diff
@@ -202,10 +202,17 @@ def delete(self, request, organization, version):
         if not self.has_release_permission(request, organization, release):
             raise ResourceDoesNotExist
 
-        # we don't want to remove the first_release metadata on the Group, and
+        # We don't want to remove the first_release metadata on the Group, and
         # while people might want to kill a release (maybe to remove files),
-        # removing the release is prevented
-        if Group.objects.filter(first_release=release).exists():
+        # removing the release is prevented if any groups still refer to it.
+        # (That said, we make an exception if said groups are already marked for
+        # deletion.)
+        if Group.objects.filter(
+            first_release=release
+        ).exclude(status__in=[
+            GroupStatus.PENDING_DELETION,
+            GroupStatus.DELETION_IN_PROGRESS
+        ]).exists():
```

Comment:
To @mattrobenolt 's point about release deletion potentially getting slow as it is a blocking operation. Should release deletion move into a task as well? Doing that would allow the web-process to return quickly and we can handle the big blocking update in the background.